pkg/transparentproxy/validate/validator_test.go

```diff
@@ -54,7 +53,8 @@ var _ = Describe("Should Validate iptables rules", func() {
 
 	It("should return fail when no iptables rules setup", func() {
 		// given
-		validator := createValidator(false)
+		validator := createValidator(false, uint16(0))
+		validator.Config.ClientConnectPort = ValidationServerPort
```

Comment:
why is this still necessary if we set the port in `validator.go`?

Reply:
Should ClientConnectPort even be configurable?

Reply:
It actually is only usable in this particular case when we want the client to connect to an invalid port. Without this, we can not control the validator to distingwish from these two scenarios:

1. auto generate a port and connect to that port (useful to test the validation logic without any iptables rules applied, this is used in the case above)
2. connect to a different port from the server listening port  (this should succed when iptables rules are applied, but it only holds in a real product run)

Reply:
Why even have the logic for 1. in `validator.go` though? It's only ever executed by the test. It's not really testing anything.

Reply:
Where is `ClientConnectPort` ever set otherwise? I don't see it in the code.

Reply:
I don't know the flow seems odd to me. `LocalServer` is mutating `Config.ClientConnectPort` so that we can later pass `ClientConnectPort` to `LocalClient`, it's difficult to follow what's happening here.

Reply:
I would expect something like

1. run Server(porttolistenon), either 0 or some port, which returns the port the server is listening on
2. run Client(porttoconnectto)

Between 1 and 2 we have to decide whether we want to use `porttolistenon` or a random port but we should only have code to do the first in the tests and only execute "find random port to connect to" in the real world.
The interface shouldn't know that it's in test mode or not test mode IMO.